frontend/src/components/crop/Plot.tsx

```diff
@@ -21,6 +28,25 @@ export default function ImagePlot({ image, max_pixel_value }: ImagePlotProps) {
           yLabel: "y-axis",
         }}
         values={image}
+        selectionsListener={(eventType, dragging, selection) => {
+          if (selection != undefined) {
+            if (eventType == "created") {
+              if (selection instanceof RectangularSelection) {
+                setCurrentSelection([selection]);
+              } else {
+                // copy the value of currentSelection and set it to that again (dont change it)
+                // this stops regions being added if theyre not a rectangle
+                // however, the component still needs to refresh as the new selection region will be visible otherwise
+                // lmk if theres a better way to "force refresh" a component
```

Comment:
Yep, I see what you mean: if a non-rectangle selection is drawn then we need to keep the current selection (whether it be a rectangle selection that already exists, or no selection exists), but we _also_ need to make sure a rerender occurs to have the non-rectangle selection that was attempted to be drawn be cleared.

I think what you have done is the react way to achieve this: trigger a rerender by setting the state to some new data object (it's indeed an array with the same data as what was there before, but crucially, it's not the same array object, because a new array has been constructed, which causes the rerender).

Therefore, this seems good to me, and the final line of the comment can be removed :slightly_smiling_face: